compiler/modules/comm.diagnostic/diagnostic.bal

```diff
@@ -110,6 +113,92 @@ public function messageFormat(Template t) returns string {
     return string:concat(...strs);
 }
 
+public type DiagnosticError error<Diagnostic>;
+
+public type DiagnosticPrinter object {
+    public function addDiagnostic(DiagnosticError diagnostic);
+    public function flushDiagnostics() returns (file:Error|io:Error)?;
+};
+
+public class StdErrorPrinter {
+    *DiagnosticPrinter;
+    private DiagnosticError[] diagnostics = [];
+
+    public function addDiagnostic(DiagnosticError diagnostic) {
+       self.diagnostics.push(diagnostic);
+    }
+
+    public function flushDiagnostics() returns (file:Error|io:Error)? {
+        foreach DiagnosticError err in self.diagnostics {
+            foreach string line in format(err.detail()) {
+                io:fprintln(io:stderr, line);
+            }
+        }
+    }
+}
+
+public class HTMLErrorPrinter {
+    *DiagnosticPrinter;
+    private DiagnosticError[] diagnostics = [];
+    string[] body = [];
+    public function addDiagnostic(DiagnosticError diagnostic) {
+       self.diagnostics.push(diagnostic);
+    }
+
+    public function flushDiagnostics() returns (file:Error|io:Error)? {
+        self.body = [];
+        self.addErrorReportPrefix();
+        foreach DiagnosticError err in self.diagnostics {
+            check self.addErrorToReport(err);
+        }
+        self.addErrorReportSuffix();
+        check io:fileWriteLines("errorLog.html", self.body);
```

Comment:
Add a constructor that takes a filename.

Reply:
fixed